libtransmission/rpc-server.cc

```diff
@@ -643,6 +650,44 @@ void rpc_server_start_retry_cancel(tr_rpc_server* server)
     server->start_retry_counter = 0;
 }
 
+#ifdef WITH_LIBEVENT_OPENSSL
+struct bufferevent* createSSLBufferEvent(struct event_base* base, void* arg)
+{
+    struct bufferevent* ret = nullptr;
+    auto ctx = reinterpret_cast<SSL_CTX*>(arg);
+    ret = bufferevent_openssl_socket_new(base, -1, SSL_new(ctx), BUFFEREVENT_SSL_ACCEPTING, BEV_OPT_CLOSE_ON_FREE);
+    return ret;
+}
+
+SSL_CTX* tr_set_cert(char const* cert, char const* key)
+{
+    SSL_CTX* ctx = SSL_CTX_new(SSLv23_server_method());
```

Comment:
OpenSSL 3.x docs say[^1] that
> <dl><dt>SSLv23_method(), SSLv23_server_method(), SSLv23_client_method()</dt>
> <dd>These functions do not exist anymore, they have been renamed to TLS_method(), TLS_server_method() and TLS_client_method() respectively. Currently, the old function calls are renamed to the corresponding new ones by preprocessor macros, to ensure that existing code which uses the old function names still compiles. However, using the old function names is deprecated and new code should call the new functions instead.</dd></dl>

These methods seem to be available since at least OpenSSL 1.1 (which is our current minimum), so please use the non-deprecated name: https://github.com/transmission/transmission/blob/b565e076a95b28f46fbd6b1d804ea992bacd8e3e/CMakeLists.txt#L53

[^1]: https://www.openssl.org/docs/man3.0/man3/SSLv23_method.html